src/App.tsx

```diff
@@ -15,37 +18,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortMethod {
+  Alphabetically,
+  ByLength,
+}
+
+type Direction = 'straight' | 'reverse';
+
+function sortList(
+  list: Good[],
+  sortMethod: SortMethod | undefined,
+  direction: Direction,
+): Good[] {
+  let sortedList: Good[];
+
+  switch (sortMethod) {
+    case SortMethod.Alphabetically:
+      sortedList = [...list].sort((a: Good, b: Good) => a.localeCompare(b));
+      break;
+    case SortMethod.ByLength:
+      sortedList = [...list].sort((a: Good, b: Good) => a.length - b.length);
+      break;
+    default:
+      sortedList = [...list];
+  }
+
+  return direction === 'reverse' ? sortedList.reverse() : sortedList;
+}
+
 export const App: React.FC = () => {
+  const [sortMethod, setSortMethod] = useState<SortMethod | undefined>(
+    undefined,
+  );
+
+  const [direction, setDirection] = useState<Direction>('straight');
+
+  function resetSort() {
+    setSortMethod(undefined);
+    setDirection('straight');
+  }
+
+  const sortedGoods: Good[] = sortList(goodsFromServer, sortMethod, direction);
```

Comment:
```suggestion
  const sortedGoods = sortList(goodsFromServer, sortMethod, direction);
```

Here type is not needed, it will be set dynamically